core/src/main/java/io/grpc/ClientCallImpl.java

```diff
@@ -43,25 +47,44 @@
 import io.grpc.transport.HttpUtil;
 
 import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.ScheduledFuture;
 import java.util.concurrent.TimeUnit;
 
+import javax.annotation.concurrent.GuardedBy;
+
 /**
  * Implementation of {@link ClientCall}.
  */
 final class ClientCallImpl<ReqT, RespT> extends ClientCall<ReqT, RespT> {
+  /**
+   * Parameters from https://github.com/grpc/grpc/blob/master/doc/connection-backoff.md
+   */
+  private static final double INITIAL_BACKOFF_MILLIS = 1 * 1000;
+  private static final double MULTIPLIER = 1.6;
+  private static final double MAX_BACKOFF_MILLIS = 2 * 60 * 1000;
+  private static final double MIN_CONNECT_TIMEOUT_MILLIS = 20 * 1000;
+  private static final double JITTER = .2;
+
   private final MethodDescriptor<ReqT, RespT> method;
   private final SerializingExecutor callExecutor;
   private final boolean unaryRequest;
   private final CallOptions callOptions;
+  private ClientCall.Listener<?> callListener;
+  @GuardedBy("this")
```

Comment:
Maybe guard with a separate lock object? Probably not a good idea to synchronize on an object that is accessible to the application.